internal/rbd/rbd_util.go

```diff
@@ -1215,7 +1215,7 @@ func GenVolFromVolID(
 
 	vol, err = generateVolumeFromVolumeID(ctx, volumeID, vi, cr, secrets)
 	if !errors.Is(err, util.ErrKeyNotFound) && !errors.Is(err, util.ErrPoolNotFound) &&
-		!errors.Is(err, ErrImageNotFound) {
+		!errors.Is(err, ErrImageNotFound) && !errors.Is(err, rados.ErrPermissionDenied) {
```

Comment:
What is the reason to continue in case of _permission denied_? Can the volume be used for anything without proper permissions?

Reply:
we need to consider this error message for the case where the ceph user has restricted access to a particular pool/radosnamespace and we are not able to loop up across all the corresponding clusterID mappings what we have in the cephcsi.